packages/react-button/src/components/Button/Converged.stories.tsx

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import { Button } from './Button';
+import { UploadIcon } from '@fluentui/react-icons-mdl2';
+import { ThemeProvider } from '@fluentui/react-theme-provider';
+import { PartialTheme } from '@fluentui/theme';
+import { Stack, Text, ColorPicker, IColor } from '@fluentui/react';
+
+const getThemes = (brand: string) => {
+  const lightTheme: PartialTheme = {
+    tokens: {
+      color: {
+        body: {
+          background: 'white',
+          contentColor: 'black',
+        },
+
+        brand: {
+          background: brand,
+
+          hovered: {
+            background: brand,
+          },
+          pressed: {
+            background: brand,
+          },
+        },
+      },
+    },
+  };
+
+  const darkTheme: PartialTheme = {
+    tokens: {
+      color: {
+        body: {
+          background: '#333',
+          contentColor: 'white',
+        },
+
+        brand: {
+          background: brand,
+
+          hovered: {
+            background: '#555',
+          },
+          pressed: {
+            background: '#555',
+          },
+        },
+      },
+    },
+  };
+
+  return { lightTheme, darkTheme };
+};
+
+const ExampleBox = ({ title, theme }: { title: string; theme: PartialTheme }) => (
+  <ThemeProvider theme={theme} style={{ padding: 20 }}>
+    <Stack
+      tokens={{
+        childrenGap: 8,
+        maxWidth: 400,
+      }}
+    >
+      <Text variant="xLarge">{title}</Text>
+      <Button icon={<UploadIcon />}>I am a button</Button>
+      <Button primary>I am a primary button</Button>
+    </Stack>
+  </ThemeProvider>
+);
+
+export const ThemeExample = () => {
+  const [color, setColor] = React.useState<string>('#f00');
+  const { lightTheme, darkTheme } = getThemes(color);
+  const onColorChange = React.useCallback((ev: never, newColor: IColor) => setColor(newColor.str), []);
+
+  return (
+    <Stack gap={16}>
```

Comment:
Justin is right, the reason we split the stories/examples into a separate package is because having them in the component packages can create circular dependency issues. It doesn't happen in this case because react-button currently depends on react-internal, but it becomes a big issue once you remove the react-internal dep. The cases he mentioned are good examples. 

The possibilities I can think of to address the circular dep issue are: 
1. Put all the examples/stories in a single package, like we have today (downside: not ideal dev experience) 
2. For each component package, make a corresponding examples/stories package which can depend on anything (downside: each separate package adds maintenance and build overhead) 
3. Separate the concepts of dev-time examples (stories?) and documentation examples. Dev-time examples live in component packages (and can't depend on other components, at least not in circular ways) and documentation examples live in a single separate package. Historically we've used the same examples for development and documentation (IIRC in both v8 and v0) but maybe we should reconsider.

> That being said, maybe it's a crazy idea, but if we could avoid/solve the dependency problem (or maybe it's not a problem?). Then packages export their stories, which would make pulling examples from our components and community work nicely in the future. I like the idea of a component package (or set of packages) being self contained, should lead to better mix and matching of components in the future.

I think the only way to keep the stories in the same package and avoid the dependency issue would be option 3 above, and it would be dev-time stories only. But even if we did that, I don't think packages should export (or probably even publish) stories. Certainly shouldn't export them by default since that would lead to a confusing dev experience. 

This actually reminds me of another minor reason that we put the examples in a separate package: so they could theoretically be consumed in other published packages without inflating the size of the main package. In the past, when our examples lived in OUFR, they were also published with OUFR (because the website package uses the examples, and in the past the website package was published and partially reused by an internal version of the site). This led to complaints from some partners (ODSP mainly) that our package was too big. While this wasn't the main motivation for moving the examples, it didn't hurt.

I'd *like* to say that now that the website package isn't published, it would be possible to have examples in the component package (ignoring cycle concerns for the moment) and just exclude them from publishing using npmignore. However David mentioned on Friday that he may have to start publishing the website package again (something related to the module federation work??)...at which point we'd be stuck publishing everything it depends on too, including examples.

Reply:
Yeah, maybe I'm over complicating things with the idea of exporting stories. Perhaps we should just go with a simple setup. Or more likely this is a problem to be solved through whatever storybook setup we go with. 

I could possibly even be convinced that doc examples and dev time stories **should** be different.

Reply:
+1 on the circular dependencies problem.

I think that it is better to have just one set of examples for all - dev, docs and screenshot tests.
Also as we plan to provide other teams with FUI storybook with all components, I would prefer to have just one storybook for all components instead of having per-component storybooks.
cc @levithomason as this is closely related/overlaps with his current work

Reply:
**circular deps:**

First of all, thanks a lot for thorough background regarding this topic, which brings me back to "crucial need for requirements" :). if someone can handle us such a document, I guess we would save a lot of time.

- If we gonna use storybook, there is no such a thing as circular deps, as SB will traverse paths and bundle only files we define to bundle (`.stories.*` in this case) **to create an App not a package**. This fits right in with what is  @levithomason trying to achieve (SB composition from various repos).
- exporting stories as public API/package ?  This doesn't make any sense to me TBH (from storybook API point of view - your tool queries for what it needs to build )

**e2e coverage:** from my experience e2e should be completely separated from implementation
- `package + stories` -> storybook app available at url
- `package-e2e` -> separate folder/package (not publishable) contains only test suites that access storybook via url, thus testing it end to end

> Also as we plan to provide other teams with FUI storybook with all components, I would prefer to have just one storybook for all components instead of having per-component storybooks.

This covers both scenarios. 
- Ship+build all to one storybook/ship+build only per package. 
- Sidenote: from my experience it is rather annoying  if one needs to build whole suite to be able to work on 1 component.

Reply:
OK, this should be simple :) Seems we all agree the "ideal" scenario is to colocate stories with components for optimal dev experience and to ship those examples to the docs for our users.

Martin, show us no circular dependencies 👍

Reply:
> * If we gonna use storybook, there is no such a thing as circular deps, as SB will traverse paths and bundle only files we define to bundle (`.stories.*` in this case) **to create an App not a package**. This fits right in with what is  @levithomason trying to achieve (SB composition from various repos).

I'm confused...how does just the fact of using storybook mean there are suddenly no circular deps? Would story files in individual component packages be excluded from the package's build and tsconfig? Where in the build process would they get type checked (important to prevent stories' API usage from getting out of date)?

I'd also appreciate more background from @levithomason about this proposal of multi-repo storybook, especially whether it's also intended to be used as the inner loop for this repo (which wouldn't make sense to me...seems way too heavy-weight for inner dev loop).